### ISSUE TITLE
Any Plan to release the source code of EMNLP2021'CPFT

### ISSUE THREAD

**Issue by Doragd** [jianguoz/Few-Shot-Intent-Detection#1]
Hi, Jian Guo,
Thanks for your excellent work about few-shot learning. It is really interesting and insightful. 
I would like to know if there is a chance I can access your source code in order to study such interesting work's details.
Do you have any plan to release the source code of CPFT?

**Comment by jianguoz** [comment]
Hi @Doragd. Thanks very much for your interest. Recently, I have been busy with the full-time job interviewing process and will consider polishing the code after graduation and after the company approval.  One thing is that I previously wrote my code based on [DNNC](https://github.com/salesforce/DNNC-few-shot-intent) (there is also a pre-training phase in DNNC) and HuggingFace. Recently I found that the [codebase](https://github.com/princeton-nlp/SimCSE) of SimCSE provides great code for contrastive learning, I will rebuild my code based on them, and one can also quickly reproduce the idea based on the SimCSE codebase.

**Comment by Doragd** [comment]
@jianguoz Thanks for your quick and kind reply. I'm glad to see that the codebase will release in the near future. By the way, wish you successful graduation and find your ideal job.

**Comment by jianguoz** [comment]
> @jianguoz Thanks for your quick and kind reply. I'm glad to see that the codebase will release in the near future. By the way, wish you successful graduation and find your ideal job.

(Two many interviews make a man tired) Thanks very much:).

**Comment by Doragd** [comment]
Hi, jianguo,
Any progress on open source the [CPFT](https://arxiv.org/pdf/2109.06349.pdf) code?
(Congrats to you as the Research Scientist in Saleforces.

**Comment by jianguoz** [comment]
@Doragd Hi Gordon, thanks very much for your message! I am working on a new company with new priorities and Mainly avoiding any associated patent in filling legal issues with the code between my previous company (Adobe) and Salesforce.   Currently, I do not a promising date for your question. Thanks very much:)

**Comment by hdzhang-code** [comment]
Hi, Jianguo,

Congratulations to you as the Research Scientist in Saleforces.

It is highly appreciated if you can open source the CPFT code. I fail to reproduce the results in the paper. Is there any critical training details, such as learning rate scheduler?

Best Regards,
Haode

**Comment by hdzhang-code** [comment]
Hi, Jianguo,

Congratulations to you as the Research Scientist in Saleforces.

It is highly appreciated if you can open source the CPFT code. I fail to reproduce the results in the paper. Is there any critical training details, such as learning rate scheduler?

Best Regards,
